### PR TITLE
Restructure the project to use smaller packages and have objects be responsible for some of them

### PR DESCRIPTION
**Pull request by puff-fan-420**
This MR restructures the packages to have some code hierarchy and relieve tech debt incurred from the original release.
The changes in short:
* `server` package no longer barebones and contains the logic for initializing and routing the HTTP server
* common functions moved to `server/srvcommon`, the name of this package is definitely up for debate
* `Provider` interface has been split into three smaller interfaces that make up `Provider` once embedded, moved to `srvcommon` as well
* Base-level endpoints are controlled by the `server` package and its resident `Server` object, authorization related endpoints are in the `server/auth.go` file, but still belong to `Server`
* `verifyDomain` -> `ValidateDomain` and moved to its own package at `server/domain`, though writing this now I see no reason why it shouldn't be in `server/srvcommon`
* Added some godoc-comments, not enough
* Moved RNG seeding to `init`

Suggestion: hostname as part of env vars

A lot of the decisions in this MR are either subjective or weakly-held, so feel free to challenge.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1,486 +1,30 @@
...
-		pk := pem.EncodeToMemory(&pem.Block{
-			Type:    "RSA PRIVATE KEY",
-			Headers: nil,
-			Bytes:   x509.MarshalPKCS1PrivateKey(session.Get("privateKey").(*rsa.PrivateKey)),
-		})
-
-		return respondWithHTML(ctx, fmt.Sprintf(templates.Done, session.Get("hostname").(string), *ipv4, string(pk), string(pk)))
-	})
+	providers := map[string]srvcommon.Provider{
+		"digitalocean": &digitalocean.DigitalOcean{},
```
Comment (MissPasture) [inline main.go:23]: should be `digitalocean.New()` so we initialize `DigitalOcean.cl`

File: server/srvcommon/srvcommon.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
+// Package srvcommon contains various constants, interfaces and
```
Comment (MissPasture) [inline server/srvcommon/srvcommon.go:1]: Not a fan of this name but it'll do for now

**Review by MissPasture — CHANGES_REQUESTED** [review]

File: server/srvcommon/srvcommon.go (hunk 2 above)
Comment (puff-fan-420) [reply server/srvcommon/srvcommon.go:1]: Neither am I to be perfectly frank -- if you've got suggestions I'm happy to change it

File: main.go (hunk 1 above)
Comment (puff-fan-420) [reply main.go]: Fixed in latest commit, please verify

**Review by MissPasture — APPROVED** [review]

**Pull request merged**